usagov/usagov-2021
# Usagov 1856 datalayer twig

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1856

## Description

Moves building the datalayer items out of TWIG templates into the preprocess hook used to gather the node related information. Main change is directly getting the breadcrumb items the block accessed but before any of the items were rendered. Replicates the same logic that was in the twig layer in a DatalayerBuilder object.

## Type of Changes

  - Add screenshot showing what it should look like
    -  Refactoring which shouldn't have any side effects.

## Testing Instructions



See documentation in Datalayer_for_GTM.md for how to run the verify script to check that there are no changes in the output.

### Change Requirements



### Validation Steps

- Verify script should flag that there are no unexpected changes in the datalayer across all exported pages.

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
